docs/core/tools/dotnet-new-list.md

```diff
@@ -0,0 +1,109 @@
+---
+title: dotnet new --list option
+description: The dotnet new --list option lists available templates.
+ms.date: 04/29/2021
+---
+# dotnet new --list option
+
+**This article applies to:** ✔️ .NET Core 2.0 SDK and later versions
+
+## Name
+
+`dotnet new --list` -  Lists available templates to be run using `dotnet new`.
+
+## Synopsis
+
+```dotnetcli
+dotnet new [<TEMPLATE_NAME>] -l|--list [--author <AUTHOR>] [-lang|--language {"C#"|"F#"|VB}]
+    [--tag <TAG>] [--type <TYPE>] [--columns <COLUMNS>] [--columns-all]
+```
+
+## Description
+
+The `dotnet new --list` option lists available templates to use with `dotnet new`. If the <TEMPLATE_NAME> is specified, lists templates containing the specified name. This option lists only default and installed templates. To find templates in NuGet that you can install locally, use the [`--search`](dotnet-new-search.md) option.
+
+## Arguments
+
+- **`TEMPLATE_NAME`**
+
+  If the argument is specified, only the templates containing `<TEMPLATE_NAME>` in template name or short name will be shown.
+
+## Options
+
+- **`--author <AUTHOR>`**
+
+  Filters templates based on template author. Partial match is supported. Available since .NET Core 5.0.300 SDK.
+
+- **`--columns <COLUMNS>`**
+
+  Comma-separated list of columns to display in the output. The supported columns are:
+  - `language` - A comma-separated list of languages supported by the template.
+  - `tags` - The list of template tags.
+  - `author` - The template author.
+  - `type` - The template type: project or item.
+  
+  The template name and short name are always shown. The default list of columns is template name, short name, language, and tags. This list is equivalent to specifying `--columns=language,tags`.
+  Available since .NET Core 5.0.300 SDK.
+
+- **`--columns-all`**
+
+  Displays all columns in the output. Available since .NET Core 5.0.300 SDK.
+
+- **`-lang|--language {C#|F#|VB}`**
+
+  Filters templates based on language supported by the template. The language accepted varies by the template. Not valid for some templates.
+
+  > [!NOTE]
+  > Some shells interpret `#` as a special character. In those cases, enclose the language parameter value in quotes. For example, `dotnet new --list --language "F#"`.
+
+- **`--tag <TAG>`**
+
+  Filters templates based on template tags. To be selected, a template must have at least one tag that exactly matches the criteria. Available since .NET Core 5.0.300 SDK.
+
+- **`--type <TYPE>`**
+
+  Filters templates based on template type. Predefined values are `project` and `item`.
```

Comment:
nit: Should we take this opportunity to include "solution"?

Reply:
I feel like we should also explain the difference between types when adding "solution" in https://docs.microsoft.com/en-us/dotnet/core/tools/custom-templates. 
`dotnet new` doesn't care about types except filtering here, whereas Visual Studio does. We probably should explain in mentioned article both `type` and `language`, and supported values for `type` given their purpose in Visual Studio.  We have https://github.com/dotnet/templating/issues/2833 for it, I suggest to address it later.